fix: Provide full and correct version in `process.version`

felixrieseberg opened the pull request:
• Outputs the full and correct version from `process.version`
• Updates the `bump_version.py` script to not break it in the future

Closes https://github.com/electron/electron/issues/11101

![screen shot 2017-11-12 at 10 29 26 pm](https://user-images.githubusercontent.com/1426799/32712603-5d5e4a9e-c7fa-11e7-8a36-2618f1bf54f4.png)

zeke commented:
cc @vanessayuenn

vanessayuenn commented:
Agreed with @ckerr on point #1 and #2. @zcbenz might be able to shed light on #3 and #4. 

I also think this is a necessary patch but the change might break things for downstream users if they depend on the version number always being `x.x.x`. We'll want to make sure to communicate that clearly.

felixrieseberg commented:
@ckerr @vanessayuenn  Good points!

1) I updated the name of the variable.

2) I added a fallback

3) I don't think `ATOM_TAG` was ever used for anything public, but I could be wrong. It's certainly not used in `bump_version.py`, so if it turns out that we no longer need it, I'm happy to remove it.

4) I greped the whole codebase and found nothing _except_ for a test in `node_spec.js`. 

As for dealing with downstream fallout, I believe we can rest easy - this change would _only_ affect beta versions, while production versions will continue to report a "simple" version.

felixrieseberg commented:
@ckerr Updated once more!
